src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs

```diff
@@ -40,64 +31,53 @@ public NamingStyleCodeFixProvider()
         public override ImmutableArray<string> FixableDiagnosticIds { get; }
             = ImmutableArray.Create(IDEDiagnosticIds.NamingRuleId);
 
-        public override FixAllProvider? GetFixAllProvider()
-        {
-            // Currently Fix All is not supported for naming style violations.
-            return null;
-        }
+        public override FixAllProvider? GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
```

Comment:
❗ The batch fixer won't correctly handle naming conflicts created by parallel fixes. This is a blocking issue.

Reply:
As we sync offline, the possible case is
Task 1:  symbol-A renamed to 'B'
Task 2:  symbol-B renamed to 'A'
and it will change the semantics of the program. 

Then it means the job can't be done concurrently.
How about renaming the symbol sequentially, and when renaming this symbol cause conflicts, skip it, and do the next one.
This would at least let people can fix all naming problem if their repo doesn't contain conflicts

Reply:
Why is this not a blocking issue on the five other BatchFixers that use the rename API? Isn't this case applicable to all of them?

Reply:
Which fixers are those?  Thanks!

Reply:
I should post what I discuss with Sam offline here to make it more clearly.
What Sam was worried about is because BatchFixer computes changes in different documents separately,
If there is a document A:
```
public class INotifyFile { }
```
And also there is a document B
```
public interface iNotifyFile { }
```
Then batchFixer will generate
Document A
```
public class iNotifyFile { }
```
Document B
```
public interface INotifyFile { }
```
which caused a naming swap.

Reply:
Also technically we can avoid such naming swapping by doing some pre-check before renaming, and later continuing to rename the remaining symbols. (Also a discussion happens offline with Sam)
But it might need to do some working to inline extra works into BatchFixer before it calls [ChangeSolutionInternalAysnc](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/CodeActions/CodeAction.cs,189)

The solution currently in my mind is to use [FindRenameLocations](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Rename/Renamer.cs,e5c734b23a618c86) to find all the rename locations for all the symbols. Renaming all the underlying identifier tokens using the new name.
During the renaming, we won't generate & hold independent solution snapshot like using BatchFixer.
And conflicts and error detecting would be skipped, but i feel this might not be important for CodeFixer scenario.

Tag @sharwell  @CyrusNajmabadi  to see if this is a feasible way

Reply:
@jasonmalinowski  is the expert here on the rename system.  But yes, i do feel like it would be better to compute all the information up front, and then do the fix in one swell foop.

Reply:
> Which fixers are those? Thanks!

@CyrusNajmabadi I was wrong when I said 5. The other codefix cases with the batch fixer use `FindRenameLocationsAsync` and `ResolveConflictsAsync` which should be fine (as I understand it). However, there are two cases: `AbstractMakeMethodAsynchronousCodeFixProvider` and `AbstractMakeMethodSynchronousCodeFixProvider` which both use the batch fixer and the rename apis in this way:

https://github.com/dotnet/roslyn/blob/37d9daf0e3c2327973adf00d6cd078a14e3a21dd/src/Features/Core/Portable/MakeMethodSynchronous/AbstractMakeMethodSynchronousCodeFixProvider.cs#L33

https://github.com/dotnet/roslyn/blob/37d9daf0e3c2327973adf00d6cd078a14e3a21dd/src/Features/Core/Portable/MakeMethodSynchronous/AbstractMakeMethodSynchronousCodeFixProvider.cs#L80

https://github.com/dotnet/roslyn/blob/37d9daf0e3c2327973adf00d6cd078a14e3a21dd/src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs#L30

https://github.com/dotnet/roslyn/blob/37d9daf0e3c2327973adf00d6cd078a14e3a21dd/src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs#L171

@sharwell  I assume we want to file bugs about these cases?

Reply:
@jmarolf 
Do you feel this would be a current acceptable solution? I also talked to @sharwell  about this.
In the fix all fixer, first check the naming swapping (this could be done if create a hashset and put all the before/after naming string in it), filter all of these symbols out.
Then use a similar way like the batch fixer. (Rename symbols, calculate & merge the document text diff)
The downside is this fixer becomes a 'Fix most' solution and this would be slow for large code base. (As @CyrusNajmabadi  and Sam mentioned above)

Reply:
I guess we should have a meeting about this :( From my perspective the _important_ bit is that fix all _fixes all_. It can be slow but if it's not as slow as me individually applying the fix to each diagnostic manually (what we have today) it's a productivity enhancement in my book.

Lets not get too cute here. What is the simplest design that can work? It would also be super helpful if we wrote a test case for the situation @sharwell is concerned with so it's obvious to me what the problem is.

Reply:
Sure, I am going to create a meeting and sync offline

Reply:
The conclusion of the offline discussion:
1. The impl should based on a new renaming API, which supports renaming a set of the symbol based on a single solution snapshot.
2. The fix all should only fix based on a single rule & symbol kind.